source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -170,11 +170,16 @@ void GeoipProvider::lookupInCityDb(
   if (config_->isLookupEnabledForHeader(config_->cityHeader()) ||
       config_->isLookupEnabledForHeader(config_->regionHeader()) ||
       config_->isLookupEnabledForHeader(config_->countryHeader())) {
-    ASSERT(city_db_, "Maxmind city database is not initialised for performing lookups");
     int mmdb_error;
+    MMDB_lookup_result_s mmdb_lookup_result;
+    {
+      absl::ReaderMutexLock lock(&city_db_mutex_);
+      ASSERT(city_db_, "Maxmind city database is not initialised for performing lookups");
```

Comment:
Extreme nit; asserts are more readable if the text part says how things are supposed to be rather than how they are not, e.g.

`ASSERT(city_db_, "Maxmind city database must be initialised for performing lookups");`

(that way the text says what you are asserting, rather than the inverse which reads confusingly.)

(+same for other asserts)